src/components/Filters/SchemaSieve.ts

```diff
@@ -220,6 +223,22 @@ const flattenAccumulator = (
 			}
 
 			const newKey = parentKey + delimiter + key;
+
+			// Move any required fields into flattened items in the top-level required array
+			forEach(value.required, (requiredKey) => {
+				if (!result.required) {
+					result.required = [];
+				} else {
+					// Remove any unnecessary required items pointing at parent fields
+					result.required = without(
```

Comment:
You could also use `result.required.filter` here instead of lodash

Reply:
@sradevski the `without` method is providing utility here as it allows you to shorthand the filtering function